common/utils/src/main/scala/org/apache/spark/internal/Logging.scala

```diff
@@ -96,17 +109,21 @@ trait Logging {
   }
 
   implicit class LogStringContext(val sc: StringContext) {
-    def log(args: MDC*): MessageWithContext = {
+    def log(args: Any*): MessageWithContext = {
```

Comment:
The `args` here is intended to MDC. So that we can enforce all the variables to be MDC.